rotkehlchen/chain/evm/decoding/balancer/balancer_cache.py

```diff
@@ -51,31 +52,32 @@ def query_balancer_data(
     )
     gauge_key_parts = ((str_chain_id := str(inquirer.chain_id.value)), str(version))
     pool_key_parts = (str_chain_id,)
-    with globaldb.conn.read_ctx() as cursor:
-        existing_pools = {
-            string_to_evm_address(address)
-            for address in globaldb_get_general_cache_values(
-                cursor=cursor,
-                key_parts=(cache_type, *pool_key_parts),
-            )
-        }
-        existing_gauges = {
-            string_to_evm_address(address)
-            for address in globaldb_get_general_cache_values(
-                cursor=cursor,
-                key_parts=(CacheType.BALANCER_GAUGES, *gauge_key_parts),
-            )
-        }
-
-    if latest_pools_count == len(existing_pools):
-        with globaldb.conn.write_ctx() as write_cursor:
-            globaldb_update_cache_last_ts(
-                write_cursor=write_cursor,
-                key_parts=pool_key_parts,
-                cache_type=cache_type,
-            )
-
-        return existing_pools, existing_gauges
+    if reload_all is False:
```

Comment:
i don't think this is needed here. why? there's no need to purge the cache if the count from the api is equal to what we have stored in the database, right?